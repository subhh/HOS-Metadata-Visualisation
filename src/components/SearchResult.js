import React, { useContext } from 'react'
import { Box, Text as ReText } from 'rebass'
import styled from 'styled-components'
import { StateContext } from 'state'
import useFetchSimple from 'hooks/useFetchSimple'
import { format, shadow } from 'utils'
import Link from 'components/Link'

const { NODE_ENV, REACT_APP_DEV_SERVER } = process.env

const Wrapper = styled(Box).attrs({ px: 3, py: 4, bg: 'white' })`
  border-radius: 4px;
  max-width: 200px;
  ${shadow};
`

const Text = styled(ReText).attrs({ display: 'inline-block' })``

const Count = styled(Text).attrs({
  fontSize: 1,

  color: 'blueDark',
})`
  font-weight: 700;
`

const Publications = styled(Text).attrs({
  fontSize: 1,
  color: 'greyDark',
  children: 'Publikationen',
})`
  text-transform: uppercase;
  font-weight: 500;
`

const createLink = query =>
  `${
    NODE_ENV === 'development' ? REACT_APP_DEV_SERVER : ''
  }/de/suchen-entdecken/discovery/?tx_find_find[__referrer][@extension]=Find&tx_find_find[__referrer][@vendor]=Subugoe&tx_find_find[__referrer][@controller]=Search&tx_find_find[__referrer][@action]=index&tx_find_find[__referrer][arguments]=YToxOntzOjE6InEiO2E6Nzp7czozOiJhbGwiO3M6MTU6IkRhdGEgUHJvY2Vzc2luZyI7czo3OiJhdXRob3JzIjtzOjA6IiI7czo3OiJkZWZhdWx0IjtzOjA6IiI7czo4OiJmdWxsdGV4dCI7czowOiIiO3M6MTA6InB1Ymxpc2hlcnMiO3M6MDoiIjtzOjg6InN1YmplY3RzIjtzOjA6IiI7czo2OiJ0aXRsZXMiO3M6MDoiIjt9fQ==84ceee3840c75fe70ccbde38d63fe12889b190b6&tx_find_find[__referrer][@request]=a:4:{s:10:"@extension";s:4:"Find";s:11:"@controller";s:6:"Search";s:7:"@action";s:5:"index";s:7:"@vendor";s:7:"Subugoe";}fcac206facd7fc97ccf479c8905fa76ff43930c6&tx_find_find[__trustedProperties]=a:1:{s:1:"q";a:7:{s:7:"default";i:1;s:3:"all";i:1;s:7:"authors";i:1;s:8:"fulltext";i:1;s:10:"publishers";i:1;s:8:"subjects";i:1;s:6:"titles";i:1;}}6c850a55e9c482f4b614787373e30d12a8823c5d&tx_find_find[q][default]=&tx_find_find[q][all]="${
    query ? encodeURIComponent(query) : ''
  }"&tx_find_find[q][authors]=&tx_find_find[q][fulltext]=&tx_find_find[q][publishers]=&tx_find_find[q][subjects]=&tx_find_find[q][titles]=#tx_find`

/**
 * Component displays the hit count while searching and links to the catalogue
 */
const SearchResult = () => {
  const { query } = useContext(StateContext)
  const { count, isLoading } = useFetchSimple(query)

  if (isLoading) return null

  return (
    <Wrapper>
      {count > 0 ? (
        <>
          <Box mb={3}>
            <Count mb={1}>{format(count)}</Count> <Publications />
          </Box>
          <Link href={createLink(query)}>anzeigen</Link>
        </>
      ) : (
        <ReText fontSize={1} lineHeight={1.5} color="greyDark">
          Es wurde leider keine Publikation mit dem Suchbegriff gefunden.
        </ReText>
      )}
    </Wrapper>
  )
}

export default SearchResult
