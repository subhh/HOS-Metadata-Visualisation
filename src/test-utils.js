import React, { useReducer } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from 'react-router-dom'
import { StateContext, DispatchContext, reducer, initialState } from './state'
import theme from './theme'

const AllTheProviders = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <ThemeProvider theme={theme}>
      <StateContext.Provider value={state}>
        <DispatchContext.Provider value={dispatch}>
          {children}
        </DispatchContext.Provider>
      </StateContext.Provider>
    </ThemeProvider>
  )
}

const WithRouter = ({ children }) => (
  <MemoryRouter>
    <AllTheProviders>{children}</AllTheProviders>
  </MemoryRouter>
)

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export const customRenderWithRouter = (ui, options) =>
  render(ui, {
    wrapper: WithRouter,
    ...options,
  })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const mockAsync = data => {
  jest.spyOn(global, 'fetch').mockImplementation(url => {
    return Promise.resolve({
      json: () => Promise.resolve(data),
    })
  })
}

export const restoreAsyncMock = () => {
  global.fetch.mockRestore()
}
