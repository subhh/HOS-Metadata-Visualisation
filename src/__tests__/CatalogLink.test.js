import React from 'react'
import { render, waitForElement } from 'test-utils'
import SearchResult from 'components/SearchResult'

it('<SearchResult /> renders correctly', async () => {
  const mockData = {
    response: {
      numFound: 10,
    },
  }

  const text = 'anzeigen'

  jest.spyOn(global, 'fetch').mockResolvedValue({
    json: () => Promise.resolve(mockData),
  })

  const { getByText, asFragment } = render(<SearchResult />)

  expect(asFragment()).toMatchSnapshot()

  const node = await waitForElement(() => getByText(text))

  expect(fetch).toHaveBeenCalledTimes(1)
  expect(node).toHaveTextContent(text)

  expect(asFragment()).toMatchSnapshot()

  global.fetch.mockRestore()
})
