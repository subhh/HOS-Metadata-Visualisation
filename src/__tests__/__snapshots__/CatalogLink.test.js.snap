// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> renders correctly 1`] = `<DocumentFragment />`;

exports[`<SearchResult /> renders correctly 2`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa sc-dnqmqq lortbm"
  >
    <div
      class="sc-bdVaJa hxBhjM"
    >
      <div
        class="sc-bdVaJa sc-htpNat sc-iwsKbI sc-gZMcBi iSrIHg"
        color="blueDark"
        display="inline-block"
        font-size="1"
      >
        10
      </div>
       
      <div
        class="sc-bdVaJa sc-htpNat sc-iwsKbI sc-gqjmRU kAhUWs"
        color="greyDark"
        display="inline-block"
        font-size="1"
      >
        Publikationen
      </div>
    </div>
    <a
      class="sc-bdVaJa sc-htpNat sc-ifAKCX sc-htoDjs iAeHbj"
      color="blue"
      font-size="1"
      href="/de/suchen-entdecken/discovery/?tx_find_find[__referrer][@extension]=Find&tx_find_find[__referrer][@vendor]=Subugoe&tx_find_find[__referrer][@controller]=Search&tx_find_find[__referrer][@action]=index&tx_find_find[__referrer][arguments]=YToxOntzOjE6InEiO2E6Nzp7czozOiJhbGwiO3M6MTU6IkRhdGEgUHJvY2Vzc2luZyI7czo3OiJhdXRob3JzIjtzOjA6IiI7czo3OiJkZWZhdWx0IjtzOjA6IiI7czo4OiJmdWxsdGV4dCI7czowOiIiO3M6MTA6InB1Ymxpc2hlcnMiO3M6MDoiIjtzOjg6InN1YmplY3RzIjtzOjA6IiI7czo2OiJ0aXRsZXMiO3M6MDoiIjt9fQ==84ceee3840c75fe70ccbde38d63fe12889b190b6&tx_find_find[__referrer][@request]=a:4:{s:10:\\"@extension\\";s:4:\\"Find\\";s:11:\\"@controller\\";s:6:\\"Search\\";s:7:\\"@action\\";s:5:\\"index\\";s:7:\\"@vendor\\";s:7:\\"Subugoe\\";}fcac206facd7fc97ccf479c8905fa76ff43930c6&tx_find_find[__trustedProperties]=a:1:{s:1:\\"q\\";a:7:{s:7:\\"default\\";i:1;s:3:\\"all\\";i:1;s:7:\\"authors\\";i:1;s:8:\\"fulltext\\";i:1;s:10:\\"publishers\\";i:1;s:8:\\"subjects\\";i:1;s:6:\\"titles\\";i:1;}}6c850a55e9c482f4b614787373e30d12a8823c5d&tx_find_find[q][default]=&tx_find_find[q][all]=\\"\\"&tx_find_find[q][authors]=&tx_find_find[q][fulltext]=&tx_find_find[q][publishers]=&tx_find_find[q][subjects]=&tx_find_find[q][titles]=#tx_find"
    >
      anzeigen
    </a>
  </div>
</DocumentFragment>
`;
